src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -606,14 +597,21 @@ class OpenShiftDSL implements Serializable {
                         String ns = (String)obj2MapMetadata.get("namespace");
                         if (ns == null || ns.trim().length() == 0)
                             ns = currentContext.getProject();
+                        String kind = obj2Map.get("kind");
+                        String name = obj2MapMetadata.get("name");
                         r = innerObjectDefAction(verb, obj2, userArgs, ns, r);
-                        projectNames.add(ns);
+                        projectNames.put(kind.toLowerCase().trim()+"/"+name, ns);
```

Comment:
isn't this going to have collisions if someone is creating a "build/foo" in two namespaces?

Reply:
yep you are right

alas, more groovy pain to come as I figure out how to assess :-(

Reply:
ok I think the change to address is minor .... will test out in the AM